src/main/java/seedu/address/model/person/UniquePersonList.java

```diff
@@ -116,6 +119,48 @@ public void setPersons(List<Person> persons) {
         return internalUnmodifiableList;
     }
 
+    /**
+     * Returns a list of distinct tags.
+     *
+     * @return A list of distinct tags.
+     */
+    public List<Tag> getDistinctTagList() {
+        return internalList.stream()
+                .flatMap(person -> person.getTags().stream())
+                .distinct()
+                .sorted(Comparator.comparing(tag -> tag.tagName))
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Calculates the number of students grouped under the same tags by checking the number of duplicate tags used.
```

Comment:
Yes. So for each tag in the distinct tag list, it will go through all the students in TAB to check if the student has this particular tag and increment correspondingly.